Source/WebCore/Modules/mediasource/MediaSource.cpp

```diff
@@ -234,24 +228,52 @@ void MediaSource::completeSeek()
     // 2.4.3 Seeking, ctd.
     // https://dvcs.w3.org/hg/html-media/raw-file/tip/media-source/media-source.html#mediasource-seeking
 
-    ASSERT(m_pendingSeekTime.isValid());
+    ASSERT(m_pendingSeekTarget && m_seekCompletedHandler);
 
-    ALWAYS_LOG(LOGIDENTIFIER, m_pendingSeekTime);
+    ALWAYS_LOG(LOGIDENTIFIER, m_pendingSeekTarget->time);
 
     // 2. The media element resets all decoders and initializes each one with data from the appropriate
     // initialization segment.
     // 3. The media element feeds coded frames from the active track buffers into the decoders starting
     // with the closest random access point before the new playback position.
-    MediaTime pendingSeekTime = m_pendingSeekTime;
-    m_pendingSeekTime = MediaTime::invalidTime();
-    m_private->setIsSeeking(false);
-    for (auto& sourceBuffer : *m_activeSourceBuffers)
-        sourceBuffer->seekToTime(pendingSeekTime);
+    auto seekTarget = *m_pendingSeekTarget;
+    m_pendingSeekTarget.reset();
+
+    struct CallbackAggregator final : public RefCounted<CallbackAggregator> {
```

Comment:
We already have <wtf/CallbackAggregator.h> so this should probably have a more specific name.

Reply:
I've followed the existing pattern, there are multiple CallbackAggregator object in the code, they all replicate the WTF one with some changes.
There's one in WebSiteDataStore, 3 in the NetworkProcess code, and on in CDMInstanceFairPlayStreamingAVFObjC, and a few more with a slight change in the name

Reply:
I renamed it to `SeeksCallbackAggregator`